core/src/main/java/com/linecorp/armeria/server/websocket/WebSocketServiceBuilder.java

```diff
@@ -146,10 +149,19 @@ public WebSocketServiceBuilder allowedOrigins(Iterable<String> allowedOrigins) {
         return this;
     }
 
+    /**
+     * Sets the predicate which is used to match allowed origins. The same-origin is allowed by default.
+     *
+     * @see <a href="https://datatracker.ietf.org/doc/html/rfc6455#section-10.2">Origin Considerations</a>
+     */
+    public WebSocketServiceBuilder allowedOrigins(Predicate<String> originPredicate) {
+        this.originPredicate = originPredicate;
```

Comment:
nit: rnn

Reply:
Shouldn't we also add the methods that accept a regex?

Reply:
Sure, let's add it.